integration/test/chart/migration/migration_test.go

```diff
@@ -0,0 +1,226 @@
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("adding annotation to chartconfig %#q", key.TestAppReleaseName()))
+
+		chartConfig, err := config.K8sClients.G8sClient().CoreV1alpha1().ChartConfigs("giantswarm").Get(key.TestAppReleaseName(), metav1.GetOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		annotations := map[string]string{}
+
+		if chartConfig.Annotations != nil && len(chartConfig.Annotations) > 0 {
+			annotations = chartConfig.Annotations
+		}
+
+		annotations[annotation.DeleteCustomResourceOnly] = "true"
+		chartConfig.Annotations = annotations
+
+		_, err = config.K8sClients.G8sClient().CoreV1alpha1().ChartConfigs("giantswarm").Update(chartConfig)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("added annotation to chartconfig %#q", key.TestAppReleaseName()))
+	}
+
+	// Delete chartconfig CR.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting chartconfig %#q", key.TestAppReleaseName()))
+
+		err := config.K8sClients.G8sClient().CoreV1alpha1().ChartConfigs("giantswarm").Delete(key.TestAppReleaseName(), &metav1.DeleteOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted chartconfig %#q", key.TestAppReleaseName()))
+	}
+
+	// Check chartconfig CR is deleted.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking chartconfig %#q was deleted", key.TestAppReleaseName()))
+
+		o := func() error {
+			_, err := config.K8sClients.G8sClient().CoreV1alpha1().ChartConfigs("giantswarm").Get(key.TestAppReleaseName(), metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				// Error is expected because finalizer was removed.
+				return nil
+			} else if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
+
+		n := func(err error, t time.Duration) {
+			config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("failed to get not found error: retrying in %s", t), "stack", fmt.Sprintf("%v", err))
+		}
+
+		b := backoff.NewExponential(backoff.MediumMaxWait, backoff.LongMaxInterval)
+		err := backoff.RetryNotify(o, b, n)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checked chartconfig %#q was deleted", key.TestAppReleaseName()))
+	}
+
+	// Check test-app is still deployed.
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking release %#q is deployed", key.TestAppReleaseName()))
+
+		err := config.Release.WaitForStatus(ctx, key.TestAppReleaseName(), "DEPLOYED")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q is deployed", key.TestAppReleaseName()))
+	}
+}
```

Comment:
Just to confirm; Does it mean the status of the app would be `FAILED` or something else if `chart-operator` do not have migration resources?

Reply:
Good to check. There should be a helm release called `test-app`. At the end of the test it should still have status `DEPLOYED`. 

If it got deleted or was `FAILED` then the test will fail.